src/SFA.DAS.EAS.Portal.UnitTests/Worker/EventHandlers/Commitments/CohortApprovalRequestedEventHandlerTests.cs

```diff
@@ -0,0 +1,85 @@
+    {
+        public class TestContext
+        {
+            public CohortApprovalRequestedByProviderEventHandler Sut { get; private set; }
+            public Mock<ICommandHandler<CohortApprovalRequestedCommand>> MockHandler { get; private set; }
+            public Mock<IAdapter<CohortApprovalRequestedByProvider, CohortApprovalRequestedCommand>> MockAdapter { get; private set; }
+            public Mock<IMessageHandlerContext> MockMessageHandlerContext { get; private set; }
+            public Mock<IMessageContext> MockMessageContext { get; private set; }
+
+            public TestContext()
+            {
+                MockHandler = new Mock<ICommandHandler<CohortApprovalRequestedCommand>>();
+                MockAdapter = new Mock<IAdapter<CohortApprovalRequestedByProvider, CohortApprovalRequestedCommand>>();
+                MockMessageContext = new Mock<IMessageContext>();
+                MockMessageHandlerContext = new Mock<IMessageHandlerContext>();
+                MockMessageHandlerContext
+                    .Setup(m => m.MessageHeaders)
+                    .Returns(
+                    new Dictionary<string, string>
+                    {
+                        { "NServiceBus.TimeSent", DateTime.Now.ToUniversalTime().ToString("yyyy-MM-dd HH:mm:ss:ffffff Z", CultureInfo.InvariantCulture) }
+                    });
+
+                Sut = new CohortApprovalRequestedByProviderEventHandler(MockHandler.Object, MockAdapter.Object, MockMessageContext.Object);
+            }
+        }
+
+        public class Handle : CohortApprovalRequestedEventHandlerTests
+        {
+            [Test]
+            public async Task WhenCalled_ThenTheAdapterIsCalledToConvertTheMessage()
+            {
+                // arrange
+                var testContext = new TestContext();
+                CohortApprovalRequestedByProvider @event = new CohortApprovalRequestedByProviderBuilder();
+
+                // act
+                await testContext.Sut.Handle(@event, testContext.MockMessageHandlerContext.Object);
+
+                //assert
+                testContext.MockAdapter.Verify(m => m.Convert(@event), Times.Once);
+            }
+
+            [Test]
+            public async Task WhenCalled_ThenTheCommandHandlerIsCalled()
+            {
+                // arrange
+                var testContext = new TestContext();
+                CohortApprovalRequestedByProvider @event = new CohortApprovalRequestedByProviderBuilder();
+                CohortApprovalRequestedCommand command = new CohortApprovalRequestedCommandBuilder();
+
+                testContext.MockAdapter
+                    .Setup(m => m.Convert(@event))
+                    .Returns(command);
+
+                // act
+                await testContext.Sut.Handle(@event, testContext.MockMessageHandlerContext.Object);
+
+                //assert
+                testContext.MockHandler.Verify(m => m.Handle(command, It.IsAny<CancellationToken>()), Times.Once);
```

Comment:
This verifies that `Handle` is called with the reference equals 'empty' `command `.

Ideally, we should open this up, so that we verify that the command contains the correct expected properties, as it's an implementation detail that the reference equal `command` is supplied.

If we do that, we should also not use an 'empty' command, as it is what you get by default, and we could be getting a newly constructed command, rather than the adapter converted command. In other words, we should either manually set some properties in the arrange phase (or just use AutoFixture to generate the test data).

Reply:
My bad, the command is built by the builder, so is not 'empty'. Please ignore that part of my comment.